tools/python/toolchain.bzl

```diff
@@ -242,11 +243,13 @@ def define_autodetecting_toolchain(
         toolchain = ":_autodetecting_py_runtime_pair",
         toolchain_type = ":toolchain_type",
         visibility = ["//visibility:public"],
+        exec_compatible_with = HOST_CONSTRAINTS,
```

Comment:
Is this correct? I would have expected a `target_compatible_with` constraints, similar to `java_runtime`s.

Reply:
The auto detecting toolchain is a bit funny because it basically just takes `python` from `PATH`. If you unroll the generated code, it basically does this:

```
py_runtime(
   interpreter = "foo.sh"
)
# foo.sh
py=$(which python)
exec $py $@
```

So "compatible with every platform" (for both target and exec) is technically true. Usage of this toolchain basically means you're just assuming whatever system it runs on provides python. I'm not saying doing this is a _good_ idea, its just what is historically there.

Reply:
I see, then the "no constraint" toolchain seems actually correct. The "autodetecting_*" bit threw me off as I thought the detection would apply at repository rule time.

Reply:
Happy to close this, then.